org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/preferences/PersistentModelTest.groovy

```diff
@@ -0,0 +1,136 @@
+package org.eclipse.buildship.core.preferences
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.core.runtime.Path
+import org.eclipse.jdt.core.JavaCore
+
+import org.eclipse.buildship.core.CorePlugin
+import org.eclipse.buildship.core.preferences.ModelPersistence
+import org.eclipse.buildship.core.preferences.PersistentModel
+import org.eclipse.buildship.core.preferences.internal.DefaultPersistentModel
+import org.eclipse.buildship.core.test.fixtures.WorkspaceSpecification
+
+class PersistentModelTest extends WorkspaceSpecification {
+
+    IProject project
+
+    void setup() {
+        project = newProject('sample-project')
+    }
+
+    def "Empty model has sensible defaults"() {
```

Comment:
There should be no "empty model", but null, because we could no longer make a difference between "we don't know" and "empty" otherwise.